src/status_im/contacts/db.cljs

```diff
@@ -0,0 +1,89 @@
+
+(defn contact-can-be-added? [identity]
+  (if (contacts/exists? identity)
+    (:pending? (contacts/get-by-id identity))
+    true))
+
+(defn is-address? [s]
+  (.isAddress dependencies/Web3.prototype s))
+
+(defn hex-string? [s]
+  (let [s' (if (str/starts-with? s "0x")
+             (subs s 2)
+             s)]
+    (boolean (re-matches #"(?i)[0-9a-f]+" s'))))
+
+(defn valid-length? [identity]
+  (let [length (count identity)]
+    (and
+      (hex-string? identity)
+      (or
+        (and (= 128 length) (not (str/includes? identity "0x")))
+        (and (= 130 length) (str/starts-with? identity "0x"))
+        (and (= 132 length) (str/starts-with? identity "0x04"))
+        (is-address? identity)))))
+
+(s/def ::not-empty-string (s/and string? not-empty))
+(s/def ::public-key (s/and ::not-empty-string valid-length?))
+
+;;;; DB
+
+;;Contact
+
+(s/def :contact/whisper-identity ::not-empty-string)        ;we can't validate public key, because for dapps whisper-identity is just string
+(s/def :contact/name ::not-empty-string)
+(s/def :contact/address (s/nilable is-address?))            ;nil for Dapps
+(s/def :contact/private-key (s/nilable string?))            ;nil for Dapps
+(s/def :contact/public-key (s/nilable string?))             ;nil for Dapps
+(s/def :contact/photo-path (s/nilable string?))
+(s/def :contact/status (s/nilable string?))
+
+(s/def :contact/last-updated (s/nilable int?))
+(s/def :contact/last-online (s/nilable int?))
+(s/def :contact/pending? boolean?)
+(s/def :contact/unremovable? boolean?)
+
+(s/def :contact/dapp? boolean?)
+(s/def :contact/dapp-url (s/nilable string?))
+(s/def :contact/dapp-hash (s/nilable int?))
+(s/def :contact/bot-url (s/nilable string?))
+(s/def :contact/global-command (s/nilable map?))
+(s/def :contact/commands (s/nilable (s/map-of keyword? map?)))
+(s/def :contact/responses (s/nilable (s/map-of keyword? map?)))
+(s/def :contact/commands-loaded? (s/nilable boolean?))
+(s/def :contact/subscriptions (s/nilable map?))
+(s/def :contact/debug? boolean?)                            ;true when contact added using status-dev-cli
+
+(s/def :contact/contact (allowed-keys
+                          :req-un [:contact/name :contact/whisper-identity]
+                          :opt-un [:contact/address :contact/private-key :contact/public-key :contact/photo-path
+                                   :contact/status :contact/last-updated :contact/last-online :contact/pending?
+                                   :contact/unremovable? :contact/dapp? :contact/dapp-url :contact/dapp-hash
+                                   :contact/bot-url :contact/global-command :contact/commands-loaded?
+                                   :contact/commands :contact/responses :contact/debug? :contact/subscriptions]))
+
+;;Contact list ui props
+(s/def :contact-list-ui/edit? boolean?)
+
+;;Contacts ui props
+(s/def :contacts-ui/edit? boolean?)
+
+
+(s/def :contacts/contacts (s/nilable (s/map-of ::not-empty-string :contact/contact)))
```

Comment:
let's come up with solution later. It should be easy to change because now we can find all entries by `:contacts/contacts`. It may look weird, but actually when we will use `contacts` symbol after desctructuring it will be quite convenient.